src/HaskellWorks/CabalCache/Concurrent/Fork.hs

```
@@ -4,13 +4,13 @@ import Control.Monad
 
 import qualified Control.Concurrent     as IO
 import qualified Control.Concurrent.STM as STM
+import Control.Exception (finally)
 
 forkThreadsWait :: Int -> IO () -> IO ()
 forkThreadsWait n f = do
   tDone <- STM.atomically $ STM.newTVar (0 :: Int)
   forM_ [1 .. n] $ \_ -> IO.forkIO $ do
-    f
-    STM.atomically $ STM.modifyTVar tDone (+1)
+    finally f (STM.atomically $ STM.modifyTVar tDone (+1))
```

Thanks!
@newhoggy why do we even do it instead of something like:
```
forConcurrently_ [1..n] (\_ -> f)
```
?

---

Note that this fix is not enough, I also had to mess with `runQueue`. I'm still testing

---

I am looking at `runQueue`, but it is not immediately clear for me what's going on there.
Do we need that catch because it swallows the error otherwise?

---

We need to catch because the thread should increment the `tDone` counter regardless of how it exits.